lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -4576,11 +4576,23 @@ void swift::checkFunctionActorIsolation(AbstractFunctionDecl *decl) {
       checker.diagnoseIsolationErrors();
     }
   }
+
   if (auto ctor = dyn_cast<ConstructorDecl>(decl)) {
     if (auto superInit = ctor->getSuperInitCall())
       superInit->walk(checker);
   }
 
+  if (auto nonisolatedAttr = decl->getAttrs().getAttribute<NonisolatedAttr>()) {
+    if (nonisolatedAttr->isUnsafe()) {
```

Comment:
I suggest putting this check in `AttributeChecker::visitNonisolatedAttr` and diagnosing anything that isn't a `VarDecl` with storage. Otherwise, we'll miss other declarations that `nonisolated(unsafe)` doesn't make sense on.

Reply:
ok will do 👍

Reply:
done